src/plugins/unified_search/public/filter_bar/filter_view/index.tsx

```diff
@@ -54,41 +63,50 @@ export const FilterView: FC<Props> = ({
     })} ${title}`;
   }
 
-  const badgeProps: EuiBadgeProps = {
-    title,
-    color: 'hollow',
-    iconType: 'cross',
-    iconSide: 'right',
-    closeButtonProps: {
-      // Removing tab focus on close button because the same option can be obtained through the context menu
-      // Also, we may want to add a `DEL` keyboard press functionality
-      tabIndex: -1,
-    },
-    iconOnClick,
-    iconOnClickAriaLabel: i18n.translate(
-      'unifiedSearch.filter.filterBar.filterItemBadgeIconAriaLabel',
-      {
-        defaultMessage: 'Delete {filter}',
-        values: { filter: innerText },
-      }
-    ),
-    onClick,
-    onClickAriaLabel: i18n.translate('unifiedSearch.filter.filterBar.filterItemBadgeAriaLabel', {
-      defaultMessage: 'Filter actions',
-    }),
-  };
+  // need title to be undefined so that we don't show the native tooltip - we'll use EuiTooltip instead
+  const readOnlyProps = { title: undefined, color: 'hollow', tabIndex: 0 };
+
+  const badgeProps: EuiBadgeProps = readOnly
+    ? readOnlyProps
+    : {
+        ...readOnlyProps,
+        iconType: 'cross',
+        iconSide: 'right',
+        closeButtonProps: {
+          // Removing tab focus on close button because the same option can be obtained through the context menu
+          // Also, we may want to add a `DEL` keyboard press functionality
+          tabIndex: -1,
+        },
+        iconOnClick,
+        iconOnClickAriaLabel: i18n.translate(
+          'unifiedSearch.filter.filterBar.filterItemBadgeIconAriaLabel',
+          {
+            defaultMessage: 'Delete {filter}',
+            values: { filter: innerText },
+          }
+        ),
+        onClick,
+        onClickAriaLabel: i18n.translate(
+          'unifiedSearch.filter.filterBar.filterItemBadgeAriaLabel',
+          {
+            defaultMessage: 'Filter actions',
+          }
+        ),
+      };
 
   return (
-    <EuiBadge {...badgeProps} {...rest}>
-      <span ref={ref}>
-        <FilterLabel
-          filter={filter}
-          valueLabel={valueLabel}
-          fieldLabel={fieldLabel}
-          filterLabelStatus={filterLabelStatus}
-          hideAlias={hideAlias}
-        />
-      </span>
-    </EuiBadge>
+    <EuiToolTip position="bottom" content={title}>
+      <EuiBadge {...badgeProps} {...rest}>
+        <span ref={ref}>
+          <FilterLabel
+            filter={filter}
+            valueLabel={valueLabel}
+            fieldLabel={fieldLabel}
+            filterLabelStatus={filterLabelStatus}
+            hideAlias={hideAlias}
+          />
+        </span>
+      </EuiBadge>
+    </EuiToolTip>
```

Comment:
The addition of the `EuiTooltip` here is currently causing a bug where the top nav bar filter pills are no longer being truncated on smaller screen sizes - working on a fix and will delete this comment once that's done.